packages/graphiql/src/components/GraphiQL.tsx

```diff
@@ -2084,3 +2283,179 @@ function isChildComponentType<T extends ComponentType>(
 
   return child.type === component;
 }
+
+function tabsStateEditHeadersReducer(
```

Comment:
These are what i wanted to move into toolkit, but we can do that in a different PR